### ISSUE TITLE
Use expectest for test expectations

### ISSUE THREAD

**Issue by brendanzab** [Marwes/gluon#87]
[Expectest](https://crates.io/crates/expectest) makes tests cleaner and gives better errors. I've used it before and it's quite nice! Might be something to look into!

**Comment by Marwes** [comment]
I have used something similar for JavaScript and I agree that it can give better error messages. I think there is some room for caution though. Most tests atm are just `assert_eq` which is much more familiar for most people

**Comment by brendanzab** [comment]
I guess I've just been infected with RSpec at work!

**Comment by brendanzab** [comment]
We might be also able to make our own matchers, say for unification etc..

**Comment by Marwes** [comment]
That's nice problem to have!

**Comment by Marwes** [comment]
> We might be also able to make our own matchers, say for unification etc..

That would be very useful. The current tests which assert on types with type variables are very brittle

**Comment by brendanzab** [comment]
RSpec has plenty of problems (crazy dynamic scoping for one), but it certainly makes test setup/teardown and pretty printing very nice.

**Comment by zjhmale** [comment]
@Marwes  you mean we should assert the equality between types not the strings [here](https://github.com/Marwes/gluon/blob/master/base/tests/types.rs) right?
>That would be very useful. The current tests which assert on types with type variables are very brittle

**Comment by Marwes** [comment]
Yes, since `PartialEq<Type<_, _>>`  only checks for exact equality tests like https://github.com/Marwes/gluon/blob/51b85f44b6a2939714d40daf0d6fc67704f76c6d/check/tests/pass.rs#L385 can start failing if the automatic generation of variables just change a little bit. It would be better to have an equvality check which succeed on `a -> b == a0 -> b0` (but not `a -> a == a0 -> b0`).

Since this requires some sort of state to track it, it might be best to keep that logic out of the standard equality comparison however. Either with a matcher as in `expectest` or with a newtype.

Pseudo code:
````rust
pub struct Unify<'a, I, T>(&'a Type<I, T>);
impl PartialEq<Unify<'a, I, T> for Type<I, T> { .. }

assert_eq(actual_type, Unify(expected_type));
```
